src/core.js

```diff
@@ -1288,10 +1288,20 @@ define([
                 _this.setUnsavedDuplicateNodeId(false);
             } else if (e.property === 'defaultValue' && e.mug.__className === 'Item') {
                 _this.setUnsavedDuplicateChoiceValue(false);
+            } else if (e.property === 'dataParent') {
+                // TODO Dirty write hack that needs to be replace
+                e.mug.options.isDataOnly = true;
+                _this.data.core.form.insertMug(e.mug.parentMug, e.mug, 'into');
+                e.mug.options.isDataOnly = false;
             }
 
             _this.refreshMugName(e.mug);
             _this.toggleConstraintItext(e.mug);
+        }).on('question-move', function(e) {
+            if (e.mug.spec.dataParent &&
```

Comment:
If a question is moved and the parent isn't valid reset the data parent to be undefined